### DISCUSSION TITLE
Using Enums as key property in IIndexedVector

### DISCUSSION THREAD

**Discussion by cristi16** [jamescourtney/FlatSharp#246] [Q&A]
Hi James. 

Is there any particular reason why we can't use enum fields as keys in IIndexedVector?

Thanks,
Cris

**Comment by jamescourtney** [comment]
What version are you using? Empty tables are supported on newer versions of FlatSharp. Empty tables were added in [FlatSharp 5.2](https://github.com/jamescourtney/FlatSharp/releases/tag/5.2.0).

If you're on a newer version than that, this is a bug that I'll be happy to address!

**Reply by jamescourtney** [reply]
Sorry -- got confused. I guess this was initially a different question before you changed the title. When I get emails, the original version gets sent to me.

**Reply by cristi16** [reply]
Apologies, I realized shortly after I posted the question that empty tables are in fact supported and that I'm just using an older version.

**Comment by jamescourtney** [comment]
I'll look into this for you. There's no reason that I know of in principle that enums can't be sorted vector keys, but it is something that FlatSharp goes out of its way to disallow:

```c#
        // EnumTypeModel.cs
        public override bool IsValidSortedVectorKey => false;
```

There is no technical limitation here, and the fix should be as simple as writing a bunch of tests and switching that `false` to a `true`.

I need to investigate a bit more. It's pretty likely that this is just an oversight on my part (sorry!), but I do need to confirm that Google library also supports this and there isn't any weirdness there.

**Reply by cristi16** [reply]
Thanks for looking into this. It's very much appreciated.

**Reply by jamescourtney** [reply]
Hey -- I've researched this, and `flatc` does allow enums to be key fields, but it generates C# code that doesn't compile in this case:

```
// Interestingly, if you declare 'BigEnum' after the table, you do get 
// an error from flatc saying that keys must be strings or scalars.
enum BigEnum : ulong (bit_flags) {
	A,
	B,
	C,
	D
}

table SortedVectorEnumTable {
    Value:BigEnum (key);
}
```

```c#
// Cant assign BigEnum to int
int comp = (FlatSharpTests.Oracle.BigEnum)bb.GetUlong(Table.__offset(4, bb.Length - tableOffset, bb)).CompareTo(key);
```

If you can drive a fix for this into `flatc`, I'd be happy to add support to FlatSharp. However, one of the primary ways that I validate FlatSharp is by making sure that it can interoperate with `flatc` (ie, "oracle" testing). I'm hesitant to add this before I can test interop scenarios.
